pytest/tests/mocknet/mirror.py

```diff
@@ -442,7 +441,25 @@ def run_env_cmd(args, traffic_generator, nodes):
         func = lambda node: node.neard_clear_env()
     else:
         func = lambda node: node.neard_update_env(args.key_value)
-    pmap(func, nodes + [traffic_generator])
+    targeted = nodes + to_list(traffic_generator)
+    pmap(func, targeted)
+
+
+def filter_hosts(args, traffic_generator, nodes):
+    if args.filter is not None:
+        if not re.search(args.filter, traffic_generator.name()):
```

Comment:
hmm it feels a little weird to me for the filter to apply to the traffic generator. There's only one of them and it has a very predictable name, so I don't think I would ever want to use a regex to select it or not select it. If I go `mirror.py --filter asdf start-traffic`, what should that do? The most reasonable thing I would expect is that it should start nodes that have `asdf` in the name, and then it should start the traffic generator. It would be very unexpected to me that I would get an early exit and a warning when I run that command because the traffic generator doesn't have `asdf` in its name

also this will fail with a cryptic backtrace if you run `stop-traffic` with such a filter

Reply:
This filter will not be used on a regular basis. Most of the time you will want to control all the hosts.
In the rare occasion where you want to control a part of the network, you would want to run start-traffic on subset of nodes including the traffic node.

Also, `None does not have attribute .neard_runner_stop` and the line `traffic_generator is None` is not a very cyphered message. It will make the user question why would they filter out the only node affected by that command.

Reply:
> This filter will not be used on a regular basis

I don't think that's really a good reason we shoudn't want to improve the UX. If there's something confusing, we should still fix it. If it comes up only once every couple months, then that's fine. I still really think `mirror.py --filter asdf start-traffic` should start the node with `asdf` in the name, and then start the traffic generator, but I won't press it too hard